src/components/ComputeModal.js

```diff
@@ -495,8 +498,14 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
   }
 
   const willDeletePersistentDisk = () => {
-    const { persistentDisk: existingPersistentDisk } = getExistingEnvironmentConfig()
-    return existingPersistentDisk && !canUpdatePersistentDisk()
+    const { persistentDisk: existingPersistentDisk, runtime: existingRuntime } = getExistingEnvironmentConfig()
+    return existingPersistentDisk && !canUpdatePersistentDisk() && existingRuntime.tool !== 'RStudio'
+  }
+
+  const willDeletePersistentDiskForRstudio = () => {
```

Comment:
Seems like some slight duplication.. wondering if there could be `willDeletePersistentDisk()` and `isToolRStudio()` methods to control the logic.